src/frontend/js/widgets/CourseProductItem/components/PurchaseButton/index.spec.tsx

```diff
@@ -73,13 +76,68 @@ describe('PurchaseButton', () => {
     // - SaleTunnel should not be opened
     expect(screen.queryByTestId('SaleTunnel__modal')).toBeNull();
 
+    // act is needed here because we've no way, in SaleTunnel, to check useOrder !fetching state from the DOM
     // Then user can enter into the sale tunnel and follow its 3 steps
-    await act(async () => {
-      fireEvent.click(button);
+    await act(async () => userEvent.click(button));
+
+    // - SaleTunnel should have been opened
+    expect(screen.getByTestId('SaleTunnel__modal')).toBeInTheDocument();
+  });
+
+  it.only('shows cta to open sale tunnel when remaning orders is null', async () => {
+    const product = ProductFactory({ remaining_order_count: null }).one();
```

Comment:
Can you also test that the `hasAtLeastOneCourseRun` has priority over `hasAtLeastOneRemainingOrder`.